SDK bump v0.25.0

jpculp opened the pull request:
**Description of changes:**

Various fixes, tweaks, and bumps to build with SDK v0.25.0

**Build**

- Update SDK to 0.25.0
- Add `BUILDSYS_SDK_NAME`
_(to make it easier to point to custom names)_

**buildsys**

- Allow 'dead code' in BuildPackage
_(fixes clippy error in manifest.rs)_

**grub**

- Remove `grub2-tools` as a dependency
_(no longer needed after #1893,
nor part of SDK from v0.25.0)_

**Actions**

- Bump rust to 1.58.0
- Bump cargo-make to 0.35.8

**Testing done:**

- [x] Build **aws-k8s-1.21** `x86_64` on `x86_64`
- [x] Build **aws-k8s-1.21** `aarch64` on `x86_64`
- [x] Build **aws-ecs-1** `aarch64` on `aarch64`
- [x] Build **aws-ecs-1** `x86_64` on `aarch64`


**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

jpculp committed:
buildsys: allow 'dead code' in BuildPackage

This allows us to include data within BuildPackage that is not yet in
use (ex. releases_url).

jpculp committed:
actions-workflow: bump rust to 1.58.0 and cargo-make to 0.35.8

cbgbt reviewed (approved):
LGTM

bcressey commented on Makefile.toml:
```diff
@@ -115,8 +117,8 @@ AMI_DATA_FILE_SUFFIX = "amis.json"
 # on the command line.
 
 # Depends on ${BUILDSYS_REGISTRY}, ${BUILDSYS_ARCH} and ${BUILDSYS_SDK_VERSION}.
```
nit:
```suggestion
# Depends on ${BUILDSYS_ARCH}, ${BUILDSYS_REGISTRY}, ${BUILDSYS_SDK_NAME}, and 
# ${BUILDSYS_SDK_VERSION}.
```

bcressey commented on tools/buildsys/src/manifest.rs:
```diff
@@ -220,6 +220,7 @@ struct Metadata {
 
 #[derive(Deserialize, Debug)]
 #[serde(rename_all = "kebab-case")]
+#[allow(dead_code)]
```
What is the dead code?

jpculp replied:
https://github.com/bottlerocket-os/bottlerocket/pull/1748 added release URLs to `BuildPackage`, but it isn't currently in use by any code. This allows the "dead code" only in the scope of  `BuildPackage`.

bcressey reviewed (approved):
LGTM pending final SDK

jpculp committed:
build: add BUILDSYS_SDK_NAME

This commit makes it easier point to SDKs with custom names.

jpculp committed:
grub: remove `grub2-tools` as a dependency

This package is no longer necessary as we'll use `grub-mkimage` from
build step.

jpculp committed:
build: update SDK to 0.25.0

etungsten commented on Makefile.toml:
```diff
@@ -25,10 +25,12 @@ BUILDSYS_NAME = "bottlerocket"
+BUILDSYS_SDK_VERSION="v0.25.0"
 # Site for fetching the SDK
-BUILDSYS_REGISTRY="public.ecr.aws/bottlerocket"
+BUILDSYS_REGISTRY="public.ecr.aws/c3k7p7u2"
```
I assume this is just for testing?

jpculp replied:
Yep. Once I'm finished up I'll remove that temp commit and mark this PR ready for review.
